File: php/features.php
<?php
declare(strict_types=1);

// YahooFinance SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class YahooFinanceFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new YahooFinanceBaseFeature();
            case "test":
                return new YahooFinanceTestFeature();
            default:
                return new YahooFinanceBaseFeature();
        }
    }
}

File: py/features.py
# YahooFinance SDK feature factory

from feature.base_feature import YahooFinanceBaseFeature
from feature.test_feature import YahooFinanceTestFeature


def _make_feature(name):
    features = {
        "base": lambda: YahooFinanceBaseFeature(),
        "test": lambda: YahooFinanceTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
